datalake/backend/processing/import_dataset/task.py

```diff
@@ -0,0 +1,142 @@
+import json
+import logging
+from datetime import datetime
+from http.client import responses as http_responses
+from os import environ
+from typing import Any, List, MutableMapping, Union
+from urllib.parse import urlparse
+from uuid import uuid4
+
+import boto3
+from jsonschema import ValidationError, validate  # type: ignore[import]
+from smart_open import open as smart_open  # type: ignore[import]
+
+from ..assets_model import ProcessingAssetsModel
+
+JsonList = List[Any]
+JsonObject = MutableMapping[str, Any]
+
+ssm_client = boto3.client("ssm")
+sts_client = boto3.client("sts")
+s3_client = boto3.client("s3")
+s3control_client = boto3.client("s3control")
+
+ENV = environ.get("DEPLOY_ENV", "test")
+STORAGE_BUCKET_PARAMETER = f"/{ENV}/storage-bucket-arn"
+S3_BATCH_COPY_ROLE_PARAMETER = f"/{ENV}/s3-batch-copy-role-arn"
+
+
+BODY_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "dataset_id": {"type": "string"},
+        "version_id": {"type": "string"},
+        "metadata_url": {"type": "string"},
+    },
+    "required": ["dataset_id", "metadata_url", "version_id"],
+}
+
+
+def lambda_handler(payload: JsonObject, _context: bytes) -> JsonObject:
+    """Main Lambda entry point."""
+
+    logger = set_up_logging()
+    logger.debug(json.dumps({"payload": payload}))
+
+    # validate input
+    try:
+        validate(payload, BODY_SCHEMA)
+    except ValidationError as error:
+        return error_response(400, error.message, logger)
+
+    dataset_id = payload["dataset_id"]
+    dataset_version_id = payload["version_id"]
+    metadata_url = payload["metadata_url"]
+
+    storage_bucket_arn = get_param(STORAGE_BUCKET_PARAMETER)
+    storage_bucket_name = storage_bucket_arn.rsplit(":", maxsplit=1)[-1]
+
+    staging_bucket_name = urlparse(metadata_url).netloc
+    manifest_key = f"manifests/{dataset_version_id}.csv"
+
+    with smart_open(f"s3://{storage_bucket_name}/{manifest_key}", "w") as s3_manifest:
+        for file in ProcessingAssetsModel.query(
+            f"DATASET#{dataset_id}#VERSION#{dataset_version_id}"
+        ):
+            logger.debug(json.dumps({"Adding file to manifest": file.url}, default=str))
+            key = urlparse(file.url).path[1:]
+            s3_manifest.write(f"{staging_bucket_name},{key}\n")
+
+    # trigger s3 batch copy operation
+    response = s3control_client.create_job(
+        AccountId=sts_client.get_caller_identity()["Account"],
```

Comment:
If `get_caller_identity` fails we might only get a `KeyError` message here with no more context. Something like `assert "Account" in caller_identity := sts_client.get_caller_identity(), caller_identity` (not sure if that syntax actually works) might be appropriate.

Reply:
I think it's a bit extraneous to be asserting that the AWS libraries are returning what they say they will.

Reply:
That argument is only valid if the call is *guaranteed* to succeed, which can only be the case if the call is local. This call can fail for any number of reasons, like a broken network connection, missing credentials, a change in the API, or an API outage. If we just do `AccountId=sts_client.get_caller_identity()["Account"]` we'll have no way of figuring out what went wrong if that call ever fails.